src/ports/utils.ts

```diff
@@ -57,3 +66,69 @@ export async function processRequestError(action: string, response: Response) {
     `Error ${action}, the server responded with: ${response.status}`
   )
 }
+
+export function isPaginated<T>(
+  data: T | PaginatedResponse<T>
+): data is PaginatedResponse<T> {
+  return (data as PaginatedResponse<T>).results !== undefined
+}
+
+export async function queryMultipleTimesWhenExceedingUrlLimit<T>(
```

Comment:
Would you mind adding tests for this new auxiliary function?

Reply:
Done!